components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-nodes/reporting-nodes.component.html

```diff
@@ -116,7 +116,7 @@
   <app-page-picker
     class="reporting-nodes-paging"
     [perPage]="reportData.nodesListParams.perPage"
-    [total]="reportData.nodesList.total"
+    [total]="getTotalNodes()"
```

Comment:
So here's the thing... Whenever we use a **data** function like this (as opposed to an **action** function like a click handler), Angular cannot know if its value has changed so it needs to call it. Repeatedly. I have not spent time fully characterizing how often, but in some instances I have seen it looping endlessly for no apparent reason. Since that time I have been espousing a ban on data functions in templates unless it is absolutely unavoidable. 

There are two alternative approaches: "declared properties" and "pure pipes". See my extensive notes on these in the bottom half of the preamble to https://github.com/chef/automate/pull/3810. That PR also shows examples of the "declared properties" approach, which is more appropriate in this case.